### PR TITLE
STONEBLD-694: Add e2e test to verify cachi2 pip support

### PR DESCRIPTION
**Pull request by Michkov**
Use https://github.com/cachito-testing/pip-e2e-test for testing hermetic build with pip.

### CODE REVIEW

File: hack/util-install-bundle.sh [hunk 1]
```diff
@@ -31,6 +31,19 @@ spec:
...
+        - name: hermetic
+          value: "true"
+        - name: prefetch-input
+          value: "pip"
+      pipelineRef:
+        name: docker-build
+        bundle: ${BUNDLE}
+      when:
+        dockerfile: true
+        language: Python
```
Comment (chmeliik) [inline hack/util-install-bundle.sh:45]: "python" is lowercase in https://github.com/cachito-testing/pip-e2e-test/blob/main/devfile.yaml#L3, not sure if that matters

**Review by chmeliik — COMMENTED** [review]
changes LGTM, probably won't pass until #302 is in though

File: hack/util-install-bundle.sh (hunk 1 above)
Comment (Michkov) [reply hack/util-install-bundle.sh:45]: does not matter

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Comment by Michkov** [comment]
test will fail on:
```
Pipeline build-templates-e2e/docker-build can't be Run; it contains Tasks that don't exist: Couldn't retrieve Task "sanity-inspect-image": GET https://quay.io/v2/auth?scope=repository%3Aredhat-appstudio-tekton-catalog%2Fpull-request-builds%3Apull&service=quay.io: unexpected status code 502 Bad Gateway: <html>
```

**Pull request merged**